File: backend/app/repositories/local_user_repository.py
from sqlalchemy import select
from sqlalchemy.orm import Session

from app.models import LocalUser


class LocalUserRepository:
    def get_by_id(self, db: Session, *, user_id: str) -> LocalUser | None:
        return db.get(LocalUser, user_id)

    def get_by_username(self, db: Session, *, username: str) -> LocalUser | None:
        stmt = select(LocalUser).where(LocalUser.username == username)
        return db.execute(stmt).scalars().first()

    def get_by_email(self, db: Session, *, email: str) -> LocalUser | None:
        stmt = select(LocalUser).where(LocalUser.email == email)
        return db.execute(stmt).scalars().first()

    def create(self, db: Session, *, user: LocalUser) -> LocalUser:
        db.add(user)
        return user

File: dotnet_backend/Repositories/LocalUserRepository.cs
using Microsoft.EntityFrameworkCore;
using dotnet_backend.Data;
using dotnet_backend.Entities;
using dotnet_backend.Repositories;

namespace dotnet_backend.Repositories;

public class LocalUserRepository : ILocalUserRepository
{
    private readonly AppDbContext _db;

    public LocalUserRepository(AppDbContext db)
    {
        _db = db;
    }

    public LocalUser? GetById(string userId)
    {
        return _db.LocalUsers.Find(userId);
    }

    public LocalUser? GetByUsername(string username)
    {
        return _db.LocalUsers.FirstOrDefault(u => u.Username == username);
    }

    public LocalUser? GetByEmail(string email)
    {
        return _db.LocalUsers.FirstOrDefault(u => u.Email == email);
    }

    public LocalUser Create(LocalUser user)
    {
        _db.LocalUsers.Add(user);
        _db.SaveChanges();
        return user;
    }
}

